lib/Parse/ParseDecl.cpp

```diff
@@ -4976,7 +5001,8 @@ bool Parser::isStartOfSwiftDecl(bool allowPoundIfAttributes) {
   // Otherwise, do a recursive parse.
   Parser::BacktrackingScope Backtrack(*this);
   consumeToken(tok::identifier);
-  return isStartOfSwiftDecl(/*allowPoundIfAttributes=*/false);
+  return isStartOfSwiftDecl(/*allowPoundIfAttributes=*/false,
+                            /*hadAttrsOrModifiers=*/true);
```

Comment:
What’s the attribute or modifier that has been consumed here?

Reply:
At this point `Tok` is a contextual keyword, and all decl introducers with contextual keyword (e.g. `macro`) are handled above. The all remaining cases are simple decl modifiers.

Reply:
> At this point `Tok` is a contextual keyword

We don’t *know* that it’s a contextual keyword, right? We just continue pretending that it might be, but really, it could be any identifier, right? Or am I missing something?

Reply:
There's 
```
  // If this can't possibly be a contextual keyword, then this identifier is
  // not interesting.  Bail out.
  if (!Tok.isContextualDeclKeyword())
    return false;
```
above.

Reply:
Ah, OK. I just didn’t read far enough back. It’s what you get from large functions that implicitly retain state from other early exits…

Thanks for the explanation. Good with me 👍🏽